### PR TITLE
Task 3 Full Name Method + Testing

### PR DESCRIPTION
**Pull request by Bjosx**
## Description

- Created full_name method for student to return title, first_name, middle_name and last_name as a string.
- Created a factory for the student object using Faker to generate random data.
- Created a spec for the student model and also added a test for the full_name method.

## Type of change

Please select the relevant option.

- [ ] Bug fix
- [x] New feature
- [ ] Breaking change
- [ ] Other

## Checklist

Please review the following checklist and ensure all tasks are completed before submitting the pull request:

- [x] I have performed a self-review of my own code
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Bjosx** [comment]
Not sure how this happened but in updating this branch to include the changes from task 2 I appear to have moved the deletion of the old migration file into this pull request.

File: spec/models/student_spec.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+RSpec.describe Student, type: :model do
+  let(:student) { build :student_with_data }
+  describe "#full_name" do
```
Comment (m0xfff) [inline spec/models/student_spec.rb:5]: The test case for the `#full_name` method looks good. If you decide to add those model validations you can add test coverage by doing:

```suggestion
  let(:student) { build :student_with_data }
  
  it { should validate_presence_of(:first_name) }
  it { should validate_presence_of(:last_name) }
  
  describe "#full_name" do
```

Also, it might be worth adding a test case to cover the scenario when one of the components is missing i.e. when `middle_name` is `nil`. Pretty sure I told you to only test this case last time we spoke, but I was mistaken, the spirit behind me saying that was that you don't want to get into the habit of testing every possible case because it can lead to an overwhelming amount of test cases, making your test suite difficult to maintain and time-consuming to run. 

Instead, focus on testing the most common and critical scenarios, as well as edge cases that could potentially break the functionality. This will provide a good balance between having a comprehensive test suite and maintaining its manageability. In this case, testing the scenario with a missing `middle_name` is a reasonable addition since it's a common real-world situation and directly related to the `full_name` method's behaviour.

**Review by m0xfff — CHANGES_REQUESTED** [review]
Your code looks generally quite good, but I have a few suggestions. I'll help you remove that migration & rebase onto develop during standup.

File: spec/factories/student.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
+require 'faker'
+#factory for creating an example student. using nested factory to try and follow best practice as outlined here: https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#best-practices
+FactoryBot.define do
+  factory :student do
+    #For creating a student with only the necessary data. Will need to be updated alongside validations in student model.
+    factory :least_data_student do
+      first_name { Faker::Name.first_name }
+      last_name { Faker::Name.last_name }
+    end
```
Comment (m0xfff) [inline spec/factories/student.rb:9]: You can add these attributes to the base `:student` factory, and add the rest of the attributes in a "Factory Bot Trait".

File: spec/models/student_spec.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  it { should validate_presence_of(:first_name) }
+  it { should validate_presence_of(:last_name) }
+
+  describe "#full_name" do
+    it "returns the full name of a student" do
+      expect(student.full_name).to eq "#{student.title} #{student.first_name} #{student.middle_name} #{student.last_name}" 
+    end
+    it "returns only fields which are not nil" do
+      expect(student2.full_name).to eq "#{student2.first_name} #{student2.last_name}"
```
Comment (m0xfff) [inline spec/models/student_spec.rb:15]:
```suggestion
      expect(student2.full_name).to eq "Matthew White" 
```

Comment (m0xfff) [reply spec/models/student_spec.rb:15]: And you'll need to update the definition of `:student2` to include the literal first name & last name.

**Review by m0xfff — CHANGES_REQUESTED** [review]

**Review by m0xfff — APPROVED** [review]
LGTM 👍
